packages/@webex/internal-plugin-metrics/src/new-metrics.ts

```diff
@@ -160,6 +160,10 @@ class Metrics extends WebexPlugin {
     options: SubmitClientEventOptions;
   }) {
     this.callDiagnosticLatencies.saveTimestamp(name);
+    // save the meetingId so we can use the meeting object in latency calculations if needed
+    if (options?.meetingId && !this.callDiagnosticLatencies.meetingId) {
```

Comment:
just wondering... wouldn't it be better for `saveTimestamp()` to just take meetingId as an optional parameter and have this logic of storing meetingId inside CallDiagnosticLatencies? Then you don't need the new public `setMeetingId()` method - currently it feels a bit weird that we have a setter for a public property meetingId - it's public so you could just set it without the setter, or maybe it should be private, but then you need a getter....